kube-runtime/src/controller/mod.rs

```diff
@@ -176,6 +176,38 @@ where
     })
 }
 
+/// Enqueues any mapper returned `Arc<K>` types for reconciliation
+#[cfg(feature = "unstable-runtime-subscribe")]
+fn trigger_others_shared<S, O, K, I>(
+    stream: S,
+    mapper: impl Fn(S::Ok) -> I + Sync + Send + 'static,
+    dyntype: O::DynamicType,
+) -> impl Stream<Item = Result<ReconcileRequest<K>, S::Error>>
```

Comment:
remarkably similar to `trigger_others` this. the ideas using this fn as a trait impl is starting to make sense.

Reply:
heh yeah. same for `trigger_owners`, everything is exactly the same except the input stream.